Added small screen layout for donation section

cdoremus opened the pull request:
When displayed on a cell phone the donation section on the home page takes up a large amount of vertical screen real estate. This PR compresses the donation section on a small screen. See the screen-shot below for a comparison of the current 5calls.org donation section on a small screen and the one created by this PR (edits to the `__layout.scss` file).

![donationsectioncomparisonsonasmallscreen_currentwebsite-vs-thispr](https://user-images.githubusercontent.com/942174/34462998-9523d680-ee1d-11e7-928c-3afdc845a545.png)

nickoneill commented:
Couple notes:

* Let's keep the header "5 calls is powered by donations from..." centered
* Keep the rounded corners on the buttons
* Swap the side of the button: button on the right, "supports" text on the left
* Even out the spacing so the red space below the buttons is the same height as the space between button sections

cdoremus commented:
@nickoneill I just pushed a few tweaks that I think addresses your concerns (see below).

![screenshot from 2017-12-31 22-46-49](https://user-images.githubusercontent.com/942174/34465481-a2f170ba-ee7c-11e7-814f-d5b1a804ff6d.png)
